pandas/indexes/numeric.py

```diff
@@ -22,6 +22,36 @@ class NumericIndex(Index):
     """
     _is_numeric_dtype = True
 
+    def __new__(cls, data=None, dtype=None, copy=False, name=None,
+                fastpath=False):
+
+        if fastpath:
+            return cls._simple_new(data, name=name)
+
+        if dtype and np.dtype(dtype) != cls._default_dtype:
```

Comment:
sorry, you can cast with the passed dtype (the data), then cast to the ``_defautl_dtype`` (if its different). The purpose of the dtype is to take a different path then constructing with the ``Index`` constructor. So

if I have float data, but pass ``dtype='int64'`` I want it to contrust an Int64Index.